crates/key-value/src/host_component.rs

```diff
@@ -1,17 +1,37 @@
-use crate::{Impl, KeyValueDispatch};
-use spin_app::DynamicHostComponent;
+use crate::{KeyValueDispatch, StoreManager};
+use spin_app::{App, AppComponent, DynamicHostComponent};
 use spin_core::HostComponent;
-use std::{collections::HashMap, sync::Arc};
+use std::sync::Arc;
 
 pub const KEY_VALUE_STORES_METADATA_KEY: &str = "key_value_stores";
 
+pub trait StoreManagerManager: Sync + Send {
+    fn get(&self, app: &App) -> Arc<dyn StoreManager>;
```

Comment:
It just occurred to me that this could easily take an `&AppComponent` instead, in case it needs access to metadata at that level.

Reply:
are you saying to reduce what is passed since `AppComponent` can be accessed from `App`?

Reply:
I agree. I think we want component here because we want to prepare the store for a specific component of an app

Reply:
I'll make the change.  Regarding your question: it was more the other direction -- you can easily get the `App` from an `AppComponent` if that's all you need.